common/djangoapps/util/log_sensitive.py

```diff
@@ -0,0 +1,175 @@
+"""
+Utilities for logging sensitive debug information such as authentication tokens.
+
+Usage:
+
+1. Read the warning in cli_gen_keys
+2. Generate keys using ``python3 -m common.djangoapps.util.log_sensitive gen-keys``
+   and follow the instructions it prints out
+3. When logging sensitive information, use like so::
+
+     logger.info(
+         "Received invalid auth token %s in Authorization header",
+         encrypt_for_log(token, settings.AUTH_ERRORS_DEBUG_KEYS)
+     )
+
+   This will log a message like::
+
+     Received invalid auth token ZXIzljFT7zTTuXyIB/jy4XTixL34EhI6pY0Gw4mxfFo=|IYS4JDE0S9wypALb5HfoeW9YB/CnSx35/PkspsecZyzTP1Zr8B/W+16Gfe+pRZhjBitQ66XTZk9YHwKfznQ1KA== in Authorization header
+
+4. If you need to decrypt one of these messages, save the encrypted portion
+   to file, retrieve the securely held private key, and run::
+
+     python3 -m common.djangoapps.util.log_sensitive decrypt --message-file ... --private-key-file ...
```

Comment:
1. I confirmed this worked.
2. From a usability perspective, it would have been simpler for me if `--message-file` were just `--message`. I don't think it makes sense to implement both from the start, and maybe we just leave this as-is because it is done?
3. From a usability perspective, what if private-key came from an environment variable? Note: I'm not as facile with encryption and wasn't clear exactly what you had in mind for gpg2 (below) and couldn't get it to work on devstack. If you think the current way is best, I just need to work with you to improve docs for me to get it working.

Reply:
I wrote it this way because I'm a little reticent to let even the encrypted data end up in the developer's shell history file.

If the private key needs to come from an environment variable, you could write `--private-key-file <(echo "$SOME_ENV_VAR})`.